crates/router/src/connector/trustpay/transformers.rs

```diff
@@ -0,0 +1,964 @@
+        }
+    }
+}
+
+fn get_pending_status_based_on_redirect_url(redirect_url: Option<String>) -> enums::AttemptStatus {
+    match redirect_url {
+        Some(_url) => enums::AttemptStatus::AuthenticationPending,
+        None => enums::AttemptStatus::Authorizing,
+    }
+}
+
+fn get_transaction_status(
+    payment_status: &str,
+    redirect_url: Option<String>,
+) -> CustomResult<(enums::AttemptStatus, Option<String>), errors::ConnectorError> {
+    let (is_failed, failure_message) = is_payment_failed(payment_status);
+    let pending_status = get_pending_status_based_on_redirect_url(redirect_url);
+    if payment_status == "000.200.000" {
+        Ok((pending_status, None))
+    } else if is_failed {
+        Ok((
+            enums::AttemptStatus::AuthorizationFailed,
+            Some(failure_message.to_string()),
+        ))
+    } else if is_payment_successful(payment_status)? {
+        Ok((enums::AttemptStatus::Charged, None))
+    } else {
+        Ok((pending_status, None))
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize, Eq, PartialEq, Clone)]
+pub enum TrustpayBankRedirectPaymentStatus {
+    Paid,
+    Authorized,
+    Rejected,
+    Authorizing,
+    Pending,
+}
+
+impl From<TrustpayBankRedirectPaymentStatus> for enums::AttemptStatus {
+    fn from(item: TrustpayBankRedirectPaymentStatus) -> Self {
+        match item {
+            TrustpayBankRedirectPaymentStatus::Paid => Self::Charged,
+            TrustpayBankRedirectPaymentStatus::Rejected => Self::AuthorizationFailed,
+            TrustpayBankRedirectPaymentStatus::Authorized => Self::Authorized,
+            TrustpayBankRedirectPaymentStatus::Authorizing => Self::Authorizing,
+            TrustpayBankRedirectPaymentStatus::Pending => Self::Authorizing,
+        }
+    }
+}
+
+#[derive(Default, Debug, Serialize, Deserialize, PartialEq)]
+pub struct TrustpayPaymentsResponse {
+    pub status: Option<i64>,
+    pub description: Option<String>,
+    #[serde(rename = "instanceId")]
+    pub instance_id: Option<String>,
+    #[serde(rename = "paymentStatus")]
+    pub payment_status: Option<String>,
+    #[serde(rename = "paymentDescription")]
+    pub payment_description: Option<String>,
+    #[serde(rename = "redirectUrl")]
+    pub redirect_url: Option<String>,
+    #[serde(rename = "redirectParams")]
+    pub redirect_params: Option<HashMap<String, String>>,
+    #[serde(rename = "PaymentRequestId")]
+    pub payment_request_id: Option<i64>,
+    #[serde(rename = "GatewayUrl")]
+    pub gateway_url: Option<Url>,
+    #[serde(rename = "ResultInfo")]
+    pub payment_result_info: Option<ResultInfo>,
+    #[serde(rename = "PaymentMethod")]
+    pub payment_method_response: Option<TrustpayPaymentMethod>,
+    #[serde(rename = "MerchantIdentification")]
+    pub merchant_identification_response: Option<MerchantIdentification>,
+    #[serde(rename = "PaymentInformation")]
+    pub payment_information_response: Option<BankPaymentInformationResponse>,
+}
```

Comment:
```suggestion
#[serde(rename_all = "camelCase")]
pub struct TrustpayPaymentsResponse {
    pub status: Option<i64>,
    pub description: Option<String>,
    pub instance_id: Option<String>,
    pub payment_status: Option<String>,
    pub payment_description: Option<String>,
    pub redirect_url: Option<String>,
    pub redirect_params: Option<HashMap<String, String>>,
    #[serde(rename = "PaymentRequestId")]
    pub payment_request_id: Option<i64>,
    #[serde(rename = "GatewayUrl")]
    pub gateway_url: Option<Url>,
    #[serde(rename = "ResultInfo")]
    pub payment_result_info: Option<ResultInfo>,
    #[serde(rename = "PaymentMethod")]
    pub payment_method_response: Option<TrustpayPaymentMethod>,
    #[serde(rename = "MerchantIdentification")]
    pub merchant_identification_response: Option<MerchantIdentification>,
    #[serde(rename = "PaymentInformation")]
    pub payment_information_response: Option<BankPaymentInformationResponse>,
}
```

Just wanted to confirm this, are we actually having both `camelCase` and `PascalCase` fields in the same struct?

Reply:
Now moved these to different responses (for cards, bank redirects) using serde untagged